src/MiniHallway.jl

```diff
@@ -0,0 +1,88 @@
+# Mini Hallway problem defined in http://cs.brown.edu/research/ai/pomdp/examples/mini-hall2.POMDP.
+# Original idea published in Littman, Cassandra and Kaelbling's ML-95 paper.
+
+# Basic parameters are:
+# discount: 0.950000
+# values: reward
+# states: 13
+# actions: 3
+# observations: 9
+# The rest is available at link
+
+
+struct Observation
+    no::Int
+end
+
+struct MiniHallway <: POMDP{Int, Int, Observation}
+    T::Array{SparseCat, 2}
+    R::Array{Float64}
+    O::Array{SparseCat{Array{Observation, 1}, Array{Float64, 1}}}
+    discount::Float64
+end
+
+function MiniHallway()
+    T = Array{SparseCat, 2}(undef, 13, 3)
+
+    T[1,1] = SparseCat((1), (1.)); T[1,2] = SparseCat((2), (1.)); T[1,3] = SparseCat((4), (1.))
+    T[2,1] = SparseCat((1), (1.)); T[2,2] = SparseCat((3), (1.)); T[2,3] = SparseCat((1), (1.))
+    T[3,1] = SparseCat((7), (1.)); T[3,2] = SparseCat((4), (1.)); T[3,3] = SparseCat((2), (1.))
+    T[4,1] = SparseCat((4), (1.)); T[4,2] = SparseCat((1), (1.)); T[4,3] = SparseCat((3), (1.))
+    T[5,1] = SparseCat((1), (1.)); T[5,2] = SparseCat((6), (1.)); T[5,3] = SparseCat((8), (1.))
+    T[6,1] = SparseCat((10), (1.)); T[6,2] = SparseCat((7), (1.)); T[6,3] = SparseCat((5), (1.))
+    T[7,1] = SparseCat((1), (1.)); T[7,2] = SparseCat((8), (1.)); T[7,3] = SparseCat((6), (1.))
+    T[8,1] = SparseCat((8), (1.)); T[8,2] = SparseCat((5), (1.)); T[8,3] = SparseCat((7), (1.))
+    T[9,1] = SparseCat((9), (1.)); T[9,2] = SparseCat((10), (1.)); T[9,3] = SparseCat((12), (1.))
+    T[10,1] = SparseCat((10), (1.)); T[10,2] = SparseCat((11), (1.)); T[10,3] = SparseCat((9), (1.))
+    T[11,1] = SparseCat((13), (1.)); T[11,2] = SparseCat((12), (1.)); T[11,3] = SparseCat((10), (1.))
+    T[12,1] = SparseCat((8), (1.)); T[12,2] = SparseCat((9), (1.)); T[12,3] = SparseCat((11), (1.))
+    T[13,1] = SparseCat((1:13), (0.083337, 0.083333, 0.083333, 0.083333, 0.083333, 0.083333, 0.083333, 0.083333, 0.083333, 0.083333, 0.083333, 0.083333, 0.))
```

Comment:
It seems like `Distributions.Uniform(1, 12)` would be most appropriate here.

Reply:
Hey, thanks for the recommendation!
The `Uniform` is for continuous distributions, so I used `DiscreteUniform`. It works like a charm so far!